js/digifinex.js

```diff
@@ -0,0 +1,515 @@
+                'orderTypes': {
+                },
+            },
+            'apiKey': '15cb6aad284ab8',
+            'secret': '48cad555befb5de238f6688b440a803f05cb6aad2',
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.privateGetTradePairs ();
+        const result = [];
+        const keys = Object.keys (response['data']);
+        for (let i = 0; i < keys.length; i++) {
+            const id = keys[i];
+            const pair = id.split ('_');
+            const baseId = pair[1];
+            const quoteId = pair[0];
+            const base = baseId.toUpperCase ();
+            const quote = quoteId.toUpperCase ();
+            const symbol = base + '/' + quote;
+            const market = response['data'][id];
+            const precision = {
+                'price': market[1],
+                'amount': market[0],
+            };
+            const limits = {
+                'amount': {
+                    'min': market[2],
+                    'max': undefined,
+                },
+                'price': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'cost': {
+                    'min': market[3],
+                    'max': undefined,
+                },
+            };
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': true,
+                'precision': precision,
+                'limits': limits,
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        const response = await this.privateGetMyposition ();
+        const result = { 'info': response };
+        const free = response['free'];
+        const frozen = response['frozen'];
+        const keysFree = Object.keys (free);
+        const keysFrozen = Object.keys (frozen);
+        const keys = this.arrayConcat (keysFree, keysFrozen);
+        for (let i = 0; i < keys.length; i++) {
+            const uppercase = keys[i].toUpperCase ();
+            const account = this.account ();
+            if (this.inArray (keys[i], keysFree)) {
+                account['free'] = free[keys[i]];
+            }
+            if (this.inArray (keys[i], keysFrozen)) {
+                account['used'] = frozen[keys[i]];
+            }
+            account['total'] = this.sum (account['free'] + account['used']);
+            result[uppercase] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        // limit can be up to 100, defaults to 10
+        const markets = await this.loadMarkets ();
+        const market = markets[symbol];
+        symbol = market['baseId'] + '_' + market['quoteId'];
```

Comment:
Unfortunately, this line is a violation of https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sending-market-ids, your help with rewriting it for unification is really welcome!

Reply:
I tried follow your instruction. But their symbol id naming schema has changed in V3. The V3 order book API returns symbol id in the new format.

Reply:
Oh, I see. No worries on that endpoint then.

Reply:
I tried follow your instruction. But their symbol id naming schema has changed in V3. The V3 order book API returns symbol id in the new format.